include/solvers/navier_stokes_scratch_data.h

```diff
@@ -839,14 +914,175 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
             fluid_particle_relative_velocity_at_particle_location[particle_i]
               .norm() *
             particle_properties[DEM::CFDDEMProperties::PropertiesIndex::dp] /
-            (kinematic_viscosity + DBL_MIN);
+            (kinematic_viscosity_at_particle_location[particle_i] + DBL_MIN);
         particle_i++;
       }
 
     average_fluid_particle_relative_velocity =
       average_fluid_particle_relative_velocity / particle_i;
   }
 
+  /** @brief Interpolates the filtered phase fraction value at the location of the particles.
+   * The latter values are used in calculating the density and viscosity of the
+   * fluid at the particles' locations when VOF is used.
+   *
+   * @param q_particles Quadrature type object that contains the location of the particles
+   * relative to the cell's frame of reference.
+   *
+   * @param current_filtered_solution The present value of the solution for the filtered phase fraction
+   *
+   * @param dof_handler_vof The DoFHandler of the VOF equations
+   */
+
+  template <typename VectorType>
+  void
+  calculate_filtered_vof_at_particle_location(
+    const Quadrature<dim> &q_particles,
+    const VectorType      &current_filtered_solution,
+    const DoFHandler<dim> &dof_handler_vof)
```

Comment:
I think I would just call the function calculate_vof_at_particle_location and then whether we provide the filtered or not to the function would be given by the way we call it. 

Also, could you change the order of the function arguments to be dof_handler, quadrature, then solution. It is more logical this way.